src/store/actions/api.actions.js

```diff
@@ -74,9 +79,98 @@ export const resetEndpoint = () => ({
     type: RESET_ENDPOINT,
 });
 
-export const willClone = data => ({
-    type: WILL_CLONE,
-    payload: data,
+export const willClone = data => {
+    const preparedPlugins = data.plugins.map(plugin => {
+        if (plugin.name === 'rate_limit') {
+            const pluginFromSchema = endpointSchema.plugins.filter(item => item.name === plugin.name)[0];
+            const { units } = pluginFromSchema.config.limit;
+            const policyFromSchema = pluginFromSchema.config.policy;
+            const getUpdatedLimit = limit => {
+                if (R.type(limit) === 'Object') {
+                    return {
+                        value: limit.value,
+                        unit: limit.unit,
+                        units,
+                    };
+                }
+
+                const arr = limit.split('-');
+                const valueOfLimit = arr[0]*1;
+                const valueOfUnit = arr[1];
+
+                return {
+                    value: valueOfLimit,
+                    unit: valueOfUnit,
+                    units,
+                };
+            };
+
+            // set the path for the lens
+            const lens = R.lensPath(['config', 'limit']);
+            const lens2 = R.lensPath(['config', 'policy']);
+            const lens3 = R.lensPath(['config', 'policy', 'selected']);
+            // substitude the plugin.config.limit
+            const updatedPlugin = R.set(lens, getUpdatedLimit(plugin.config.limit), plugin);
+            const pluginWithPolicyFromSchema = R.set(lens2, policyFromSchema , updatedPlugin);
+            const getSelectedPolicy = policy => {
+                if (R.type(policy) === 'Object') {
+                    return policy.selected;
+                }
+                return policy;
+            };
+
+            return R.set(lens3, getSelectedPolicy(plugin.config.policy), pluginWithPolicyFromSchema);
+        }
+        if (plugin.name === 'request_transformer') {
```

Comment:
Maybe we could use a `switch` statement

Reply:
Yeap, I'll do this in a separate **PR** in which I will also try to extract the common parts. This one is already too big :mask: